transport-native-io_uring/src/main/java/io/netty/channel/uring/AbstractIOUringChannel.java

```diff
@@ -714,10 +714,14 @@ protected void doRegister() throws Exception {
     }
 
     @Override
-    protected void doDeregister() throws Exception {
+    protected void doDeregister() {
         IOUringSubmissionQueue submissionQueue = submissionQueue();
 
         if (submissionQueue != null) {
+            if ((ioState & (POLL_IN_SCHEDULED | POLL_OUT_SCHEDULED | POLL_RDHUP_SCHEDULED)) == 0) {
```

Comment:
@normanmaurer we could just use `ioState == 0`, I'm not quite sure.. WDYT?

Reply:
Yeah I think that is right...